doc/tutorial.rst

```diff
@@ -0,0 +1,388 @@
+
+- ``--extra-index-url https://test.pypi.org/simple/`` is necessary to get the
+  latest version of ``Ragger``, as it is currently not deployed on the
+  stable ``pypi`` repositories.
+- ``ragger[speculos]`` means than Speculos and its dependencies will also
+  be installed. ``Ragger`` tries to uncouple its dependencies so that only
+  what is needed is installed.
+  All the extras can be seen
+  `in the setup.cfg <https://github.com/LedgerHQ/ragger/blob/develop/setup.cfg#L36-L100>`_
+  file.
+
+First backend instantiation
+---------------------------
+
+The ``SpeculosBackend``
++++++++++++++++++++++++
+
+The :py:class:`BackendInterface <ragger.backend.BackendInterface>` really only
+needs a :py:class:`Firmware <ragger.firmware.Firmware>` - declaring the type of
+Ledger cold wallet to run the :term:`application` on, and the version of the
+given :term:`SDK`, to be instantiated.
+
+.. code-block:: python
+  :linenos:
+
+  from ragger.firmware import Firmware
+  from ragger.backend import BackendInterface
+
+  backend = BackendInterface(Firmware("nanos", "2.1.0"))  # this won't work, as it is an abstract interface
+
+However, as we us Speculos which is going to start the :term:`application` inside an emulator,
+we need to provide the application ELF file as an argument:
+
+.. code-block:: python
+  :linenos:
+
+  from ragger.firmware import Firmware
+  from ragger.backend import SpeculosBackend
+
+  APPLICATION_ELF = "path/to/app.elf"
+
+  backend = SpeculosBackend(APPLICATION_ELF, Firmware("nanos", "2.1.0"))
+
+And that's it, you have a working backend! Although in the case of Speculos,
+the emulator won't start immediately. You need to use a ``with`` statement
+to start it and begin to exchange APDUs or other events with you application.
+
+.. code-block:: python
+  :linenos:
+
+  with backend:
+      backend.exchange(cla=0xE0, ins=0x01, data=bytes.fromhex("DEADBEEF"))
+      backend.right_click()
+
+This syntax will ensure that the backends properly clean what needs cleaning
+once the communication is done. For Speculos, this mean stopping the whole
+emulator, for other backend it could be as simple as closing a socket.
+
+Backend as a fixture
+++++++++++++++++++++
+
+From this, we are not far away from writing a :term:`Pytest` fixture, wich could
+take place in a ``conftest.py`` file, and used in a ``test_something.py`` file:
+
+.. code-block:: python
+  :caption: conftest.py
+  :linenos:
+
+  from pytest import fixture
+  from ragger.firmware import Firmware
+  from ragger.backend import SpeculosBackend
+
+  APPLICATION_ELF = "path/to/application.elf"
+
+  @fixture
+  def backend() -> SpeculosBackend:
+      backend = SpeculosBackend(APPLICATION_ELF, Firmware("nanos", "2.1.0"))
+      with backend:
+          yield backend
+
+.. code-block:: python
+  :caption: test_something.py
+  :linenos:
+
+  def test_something(backend):
+      backend.exchange(cla=0xE0, ins=0x01, data=bytes.fromhex("DEADBEEF"))
+      backend.right_click()
```

Comment:
Dans tous les tests on renome ce backend `client`...

Reply:
Yes, not sure it is a good thing we call it "client", as it may be confusing with the actual application client, which is to the dev to develop. As this documentation makes clear difference between the two, I preferred using the "right" name.

Reply:
Fine by me, how about making a go on all existing apps, changing `client` by `backend`, and removing the current `backend` that serves no purpose ?
We should do it now before it's out of hand

Reply:
Current `backend` as a purpose as an option, could be renamed `backend_choice` or `backend_option` or whatever.

I'll update the template provided by Ragger, and I agree we should update other apps accordingly now.